contracts/modules/TransferManager/SignedTransferManager.sol

```diff
@@ -0,0 +1,158 @@
+
+    //keep tracks of the address that allows to sign messages
+    mapping(address => bool) public signers;
+
+
+    // Emit when signer stats was changed
+    event UpdateSigners(address[] _signers, bool[] _signersStats);
+
+    // Emit when a signature has been deemed invalid
+    event InvalidSignature(bytes _data);
+
+
+    /**
+     * @notice Constructor
+     * @param _securityToken Address of the security token
+     * @param _polyAddress Address of the polytoken
+     */
+    constructor (address _securityToken, address _polyAddress)
+    public
+    Module(_securityToken, _polyAddress)
+    {
+    }
+
+    /**
+     * @notice This function returns the signature of configure function
+     */
+    function getInitFunction() public pure returns (bytes4) {
+        return bytes4(0);
+    }
+
+    /**
+    * @notice function to check if a signature is still valid
+    * @param _data signature
+    */
+    function checkSignatureIsInvalid(bytes _data) public view returns(bool){
+        return invalidSignatures[_data];
+    }
+
+    /**
+    * @notice function to remove or add signer(s) onto the signer mapping
+    * @param _signers address array of signers
+    * @param _signersStats bool array of signers stats
+    */
+    function updateSigners(address[] _signers, bool[] _signersStats) public withPerm(ADMIN) {
+        require(_signers.length == _signersStats.length, "input array length does not match");
+        for(uint8 i=0; i<_signers.length; i++){
+            signers[_signers[i]] = _signersStats[i]; 
+        }
+        emit UpdateSigners(_signers, _signersStats);
+    }
+
+    /**
+    * @notice allow verify transfer with signature
+    * @param _from address transfer from
+    * @param _to address transfer to
+    * @param _amount transfer amount
+    * @param _data signature
+    * @param _isTransfer bool value of isTransfer
+    * Sig needs to be valid (not used or deemed as invalid)
+    * Signer needs to be in the signers mapping
+    */
+    function verifyTransfer(address _from, address _to, uint256 _amount, bytes _data , bool _isTransfer) public returns(Result) {
+        if (!paused) {
+
+            require(_isTransfer == false || msg.sender == securityToken, "Sender is not the owner");
+            
+            require(invalidSignatures[_data] != true, "Invalid signature - signature is either used or deemed as invalid");
+            bytes32 hash = keccak256(abi.encodePacked(this, _from, _to, _amount));
+            bytes32 prependedHash = keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", hash));
+            address signer = _recoverSignerAdd(prependedHash, _data);
+            
+            require(signers[signer] == true, "Invalid signature - signer is not on the list");
+            
+            if (signers[signer] != true){
+                return Result.NA;
+            } else {
+                invalidSignatures[_data] = true;
+                return Result.VALID;
```

Comment:
Shouldn't it return `Result.NA` ?

Reply:
what do you think about this @adamdossa ? the original script returns VALID.

Reply:
I think VALID is fine. Use-case I had in mind is an authorised entity (e.g. a signer) can provide data to whitelist a specific transaction. A bit like the MATM but using "off-chain" data instead of an on-chain list.